### PR TITLE
Feature/unify data incoming

### PR DESCRIPTION
**Pull request by DavidSouther**
This unifies feature and data under ARCHIVE_DATA_ROOT environment flag.

### CODE REVIEW

File: src/api/record/record.ts [hunk 1]
```diff
@@ -22,13 +22,13 @@ import {
...
-  public basePath: string = join(process.cwd(), 'data');
-  public dbPath: string = join(this.basePath, '.db.json');
-  public incomingPath: string = this.config.find<string>(
-    'archive.incoming',
-    'ARCHIVE_INCOMING',
-    '/var/incoming/'
+  public basePath: string = this.config.find<string>(
+    'archive.data_root',
+    'ARCHIVE_DATA_ROOT',
+    '/var/archives'
```
Comment (stanparker) [inline src/api/record/record.ts]: I have it in there as /mnt/archives

File: src/api/record/record.spec.ts [hunk 2]
```diff
@@ -234,7 +234,7 @@ describe('Record Handler', function() {
     beforeEach(function() {
       recordMap['tape-1'] = Record.fromObj(MOCK_RECORD_1);
       let associateFs = {
-        '/var/incoming': {'path1': 'Video 1', 'path2': 'Video 2'},
+        '/var/archives/incoming': {'path1': 'Video 1', 'path2': 'Video 2'},
         './data/.db.json': '{}'
```
Comment (stanparker) [inline src/api/record/record.spec.ts]: isn't the data in /mnt/archives/data now?

Comment (DavidSouther) [inline src/api/record/record.spec.ts]: There are half a dozen ways to configure it. When you install it for Q2, you have it in /mnt/archives; the program assumes `/var/archives` for testing (`/var` is commonly used in *nix for runtime data files, `/mnt` assumes the program is running somewhere with mounted file systems, which we don't need to do).

File: src/api/record/record.ts (hunk 1 above)
Comment (DavidSouther) [inline src/api/record/record.ts]: So override it by setting an environment variable `ARCHIVE_DATA_ROOT=/mnt/archives` when you start the server. (The second argument to config.find is the environment variable to use.)

**Comment by stanparker** [comment]
Okay, so I tried this out locally on my macbook.

Here's what I did.

1. I followed the command line instructions to pull down this branch
2. I moved the /data and /incoming folders to a new folder, /var/archives/
3. I ran `npm test`
4. Then I ran `npm start -- --archive.data_root=/var/archives`

I was expecting the program to find the data folder in `/var/achives/`, instead, I got this error: 

```
MacBook-Pro-2:Montana-News-Archive stanparker$ npm start -- --archive.data_root=/var/archives

 >mtna@0.0.0 start /Users/stanparker/Devel/PronghornDigital/Montana-News-Archive
 >node dist/api/app.js "--archive.data_root=/var/archives"

2016-02-25T14:53:12.241Z - info: rupert-app listening
2016-02-25T14:53:12.243Z - info: http://MacBook-Pro-2:8080/
2016-02-25T14:53:12.244Z - info: No database found, creating a new one at /Users/stanparker/Devel/PronghornDigital/Montana-News-Archive/data/.db.json
fs.js:60
      throw err;  // Forgot a callback but don't know where? Use NODE_DEBUG=fs
      ^

Error: ENOENT: no such file or directory, open '/Users/stanparker/Devel/PronghornDigital/Montana-News-Archive/data/.db.json'
    at Error (native)
```

**Comment by stanparker** [comment]
Chatted about this and found that the `npm start` command needed to be run as 

`ARCHIVE_DATA_ROOT=/var/archives npm start`

**Comment by stanparker** [comment]
Tested locally and seems to work fine. Thanks.

**Pull request merged**